src/main/clojure/clara/rules/engine.cljc

```diff
@@ -937,15 +950,19 @@
      memory
      listener
      children
-     (filter test tokens)))
+     (filter (partial test-node-matches node test {}) tokens)))
 
   (left-retract [node join-bindings tokens memory transport listener]
     (l/left-retract! listener node tokens)
     (retract-tokens transport memory listener children tokens))
 
   (get-join-keys [node] [])
 
-  (description [node] (str "TestNode -- " (:text test))))
+  (description [node] (str "TestNode -- " (:text test)))
+
+  IConditionNode
+  (get-condition-description [this]
+    (into [:test] (:constraints (meta test)))))
```

Comment:
Might be interesting to have a test that confirms that this metadata is maintained across a serialization/deserialization. 

I feel fairly confident that the metadata would survive as the entire form should be captured, but incase something changes in the future, a test to catch it would be good.

The test would likely need to be added somewhere in the durability test:
https://github.com/cerner/clara-rules/blob/main/src/test/clojure/clara/test_durability.clj

Reply:
Sure, I will add that in as we also rely serialization heavily; I'm testing it live today on our backend.

Reply:
@EthanEChristian I've added a test to ensure that the metadata is maintained across a serialization/deserialization, it worked as expected. I also tested this on a live system and worked as expected.